crates/router/src/bin/collection_runner.rs

```diff
@@ -0,0 +1,152 @@
+mod auth {
+    include!("../../tests/connectors/connector_auth.rs");
+}
+use std::{env, process::Command};
+
+use router::types::ConnectorAuthType;
+
+// Just by the name of the connector, this function generates the name of the collection
+// Example: CONNECTOR_NAME="stripe" -> OUTPUT: postman/stripe.postman_collection.json
+fn path_generation(name: &str) -> String {
+    format!("postman/{}.postman_collection.json", name)
+}
+
+// Removes double quotes
+fn trim(str: &str) -> String {
+    str.replace('\"', "")
+}
+
+// runner starts here
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    // Usage Info
+    if args.len() < 4 {
+        println!("Usage: cargo run --package router --bin collection_runner <connector_name> <base_url> <admin_api_key>");
+        return;
+    }
+
+    // Arguments
+    let connector_name = &args[1];
+    let base_url = &args[2];
+    let admin_api_key = &args[3];
+
+    // Function calls
+    let collection_path = path_generation(connector_name);
+    let auth_map = auth::ConnectorAuthenticationMap::new();
+
+    let inner_map = &auth_map.0;
+
+    // Newman runner
+    // Depending on the conditions satisfied, variables are added. Since certificates of stripe have already
+    // been added to the postman collection, those conditions are set to true and collections that have
+    // variables set up for certificate, will consider those variables and will fail.
+
+    let mut newman_command = Command::new("newman");
+
+    newman_command.arg("run");
+    newman_command.arg(&collection_path);
+
+    newman_command
+        .arg("--env-var")
+        .arg(format!("admin_api_key={}", admin_api_key));
+
+    newman_command
+        .arg("--env-var")
+        .arg(format!("baseUrl={}", base_url));
+
+    if let Some(auth_type) = inner_map.get(&connector_name.to_string()) {
+        match auth_type {
+            ConnectorAuthType::HeaderKey { api_key } => {
+                newman_command
+                    .arg("--env-var")
+                    .arg(format!("connector_api_key={}", trim(api_key)));
```

Comment:
It is necessary as of now since the response that `read_to_string()` returns is of type: "connector_name": key_type {
        api_key: "\"access_key\"",
        key1: "\"secret_key\"",
    },